### ISSUE TITLE
udevadm settle --timeout=XY returns with exit code 1

### ISSUE THREAD

**Issue by ykorman** [systemd/systemd#16086]
**systemd version the issue has been seen with**
245

**Used distribution**
Ubuntu 20.04

**Expected behaviour you didn't see**
`udevadm settle --timeout=10` return with exit code '0'

**Unexpected behaviour you saw**
`udevadm settle --timeout=10` return with exit code '1'

**Steps to reproduce the problem**
Here's a script to reproduce the problem (shamelessly stolen from [here](https://bugzilla.redhat.com/show_bug.cgi?id=1210981#c9))

```
#!/bin/bash

set -x

# export NAME and MAJOR MINOR numbers for partition where /boot is
export $(lsblk -P | awk '{ if ($0 ~ /MOUNTPOINT="\/"/) { split($1,name,"="); gsub("\"", "", name[2]); split($2,nums,"="); gsub("\"","",nums[2]); split(nums[2],res,":"); printf "NAME=%s MAJOR=%s MINOR=%s\n", name[2], res[1], res[2]; } }')

# write the test rule 
printf "ACTION==\"add\", ENV{MAJOR}==\"$MAJOR\", ENV{MINOR}==\"$MINOR\", RUN+=\"/bin/sleep 10\"\n" > /etc/udev/rules.d/99-test.rules

# reload rules
udevadm control --reload

# syntetize udev add event for the device
echo "add" > /sys$(udevadm info --query=path --name=$NAME)/uevent

udevadm settle --timeout 3 >/dev/null 2>&1
echo short settle return $?

udevadm settle --timeout 8 >/dev/null 2>&1
echo long settle return $?

rm -f /etc/udev/rules.d/99-test.rules
udevadm control --reload
```
I think that this behaviour is fine but should be documented in the manpage - if there are still events in the udev queue when the timeout completes, 'udevadm settle' returns with exit code '1'.

**Comment by arvidjaar** [comment]
> I think that this behaviour is fine 

Actually this behavior is necessary to implement --timeout=0 option (how do you know queue is empty otherwise)? Care to submit man page PR?

**Comment by ykorman** [comment]
Sure, here's a pull request for the fix: #16088
